src/include/gurt/common.h

```diff
@@ -329,13 +329,40 @@ d_realpath(const char *path, char *resolved_path) _dalloc_;
 	})
 
 /* Internal helper macros, not to be called directly by the outside caller */
-#define __D_PTHREAD(fn, x)						\
-	({								\
-		int _rc;						\
-		_rc = fn(x);						\
-		D_ASSERTF(_rc == 0, "%s rc=%d %s\n", #fn, _rc,		\
-			  strerror(_rc));				\
-		d_errno2der(_rc);					\
+#define __D_PTHREAD(fn, x)                                                                         \
+	({                                                                                         \
+		int _rc;                                                                           \
+		_rc = fn(x);                                                                       \
+		D_ASSERTF(_rc == 0, #fn "(%p) rc=%d %s\n", x, _rc, strerror(_rc));                 \
+		d_errno2der(_rc);                                                                  \
+	})
+
+#define __D_PTHREAD_TIMED(fn, fn2, x)                                                              \
+	({                                                                                         \
+		int _rc;                                                                           \
+		_rc = fn((x));                                                                     \
+		if (_rc == EBUSY) {                                                                \
+			int             delay1 = 0;                                                \
+			int             delay2 = 1;                                                \
+			int             f      = delay1 + delay2;                                  \
+			int             c      = f;                                                \
```

Comment:
am a bit worried when we use variable names like these in macro, due to a potential conflict/scope shadow. i believe we have stuck with _ prefix for those for that reason.

Reply:
Good point.  The compiler would warn of course but I'll change this.